bootstrap

```diff
@@ -761,7 +761,7 @@ rebar_uri_parse(URIString) ->
     catch
         error:undef ->
             %% Taken from rebar_uri and trimmed down.
-            case http_uri:parse(URIString, []) of
+            case rebar3_uri:parse(URIString, []) of
```

Comment:
You can't do this, because rebar3_uri is not yet compiled by the time we bootstrap. In fact, this call is literally an inlining of rebar_uri as a module _because_ of that reason, and it needs to be reversed.

Reply:
I'll revert it, then, but I fail to understand how it can work locally.

```erlang
➜  rebar3 git:(feature/parse_upcoming_dialyzer_result) ✗ rm -rf _build
➜  rebar3 git:(feature/parse_upcoming_dialyzer_result) ✗ ./bootstrap
Evaluating config script "~/rebar3/_build/default/lib/rebar/src/rebar.app.src.script"
===> Verifying dependencies...
===> Fetching bbmustache v1.10.0
...
===> Building escript for rebar...
➜  rebar3 git:(feature/parse_upcoming_dialyzer_result) ✗
```

Reply:
I've forced pushed to the previous commit.

Reply:
Oh, I see, I'm not running that code, which is why it's not complaining but it's not crashing either.